../usr/lib/python3.7/subprocess.py:883: ResourceWarning: subprocess 62088 is still running
  ResourceWarning, source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/home/student/.local/lib/python3.7/site-packages/http3/dispatch/connection_pool.py:117: ResourceWarning: unclosed <socket.socket fd=14, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=6, laddr=('127.0.0.1', 55324)>
  origin=request.url.origin, allow_connection_reuse=allow_connection_reuse
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/home/student/.local/lib/python3.7/site-packages/http3/dispatch/connection_pool.py:117: ResourceWarning: unclosed <socket.socket fd=15, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=6, laddr=('127.0.0.1', 55326)>
  origin=request.url.origin, allow_connection_reuse=allow_connection_reuse
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/home/student/.local/lib/python3.7/site-packages/http3/dispatch/connection_pool.py:117: ResourceWarning: unclosed <socket.socket fd=16, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=6, laddr=('127.0.0.1', 55328)>
  origin=request.url.origin, allow_connection_reuse=allow_connection_reuse
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/home/student/.local/lib/python3.7/site-packages/http3/dispatch/connection_pool.py:117: ResourceWarning: unclosed <socket.socket fd=17, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=6, laddr=('127.0.0.1', 55330)>
  origin=request.url.origin, allow_connection_reuse=allow_connection_reuse
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/home/student/.local/lib/python3.7/site-packages/http3/dispatch/connection_pool.py:117: ResourceWarning: unclosed <socket.socket fd=18, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=6, laddr=('127.0.0.1', 55332)>
  origin=request.url.origin, allow_connection_reuse=allow_connection_reuse
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/home/student/.local/lib/python3.7/site-packages/http3/dispatch/connection_pool.py:117: ResourceWarning: unclosed <socket.socket fd=19, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=6, laddr=('127.0.0.1', 55334)>
  origin=request.url.origin, allow_connection_reuse=allow_connection_reuse
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/home/student/.local/lib/python3.7/site-packages/http3/dispatch/connection_pool.py:117: ResourceWarning: unclosed <socket.socket fd=20, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=6, laddr=('127.0.0.1', 55336)>
  origin=request.url.origin, allow_connection_reuse=allow_connection_reuse
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/asyncio/selector_events.py:658: ResourceWarning: unclosed transport <_SelectorSocketTransport fd=9>
  source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/asyncio/selector_events.py:658: ResourceWarning: unclosed transport <_SelectorSocketTransport fd=10>
  source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/asyncio/selector_events.py:658: ResourceWarning: unclosed transport <_SelectorSocketTransport fd=11>
  source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/asyncio/selector_events.py:658: ResourceWarning: unclosed transport <_SelectorSocketTransport fd=12>
  source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/asyncio/selector_events.py:658: ResourceWarning: unclosed transport <_SelectorSocketTransport fd=13>
  source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/asyncio/selector_events.py:658: ResourceWarning: unclosed transport <_SelectorSocketTransport fd=14>
  source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/asyncio/selector_events.py:658: ResourceWarning: unclosed transport <_SelectorSocketTransport fd=15>
  source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/asyncio/selector_events.py:658: ResourceWarning: unclosed transport <_SelectorSocketTransport fd=16>
  source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/asyncio/selector_events.py:658: ResourceWarning: unclosed transport <_SelectorSocketTransport fd=17>
  source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/asyncio/selector_events.py:658: ResourceWarning: unclosed transport <_SelectorSocketTransport fd=18>
  source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/asyncio/selector_events.py:658: ResourceWarning: unclosed transport <_SelectorSocketTransport fd=19>
  source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/asyncio/selector_events.py:658: ResourceWarning: unclosed transport <_SelectorSocketTransport fd=20>
  source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/asyncio/selector_events.py:658: ResourceWarning: unclosed transport <_SelectorSocketTransport fd=21>
  source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/asyncio/selector_events.py:658: ResourceWarning: unclosed transport <_SelectorSocketTransport fd=22>
  source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/subprocess.py:883: ResourceWarning: subprocess 62119 is still running
  ResourceWarning, source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/asyncio/selector_events.py:658: ResourceWarning: unclosed transport <_SelectorSocketTransport fd=23>
  source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
../usr/lib/python3.7/subprocess.py:883: ResourceWarning: subprocess 62186 is still running
  ResourceWarning, source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
./usr/lib/python3.7/asyncio/selector_events.py:658: ResourceWarning: unclosed transport <_SelectorSocketTransport fd=27>
  source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/subprocess.py:883: ResourceWarning: subprocess 62223 is still running
  ResourceWarning, source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
../usr/lib/python3.7/asyncio/selector_events.py:658: ResourceWarning: unclosed transport <_SelectorSocketTransport fd=35>
  source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/subprocess.py:883: ResourceWarning: subprocess 62288 is still running
  ResourceWarning, source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/subprocess.py:883: ResourceWarning: subprocess 62138 is still running
  ResourceWarning, source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/subprocess.py:883: ResourceWarning: subprocess 62257 is still running
  ResourceWarning, source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/asyncio/selector_events.py:658: ResourceWarning: unclosed transport <_SelectorSocketTransport fd=25>
  source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/asyncio/selector_events.py:658: ResourceWarning: unclosed transport <_SelectorSocketTransport fd=24>
  source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/asyncio/selector_events.py:658: ResourceWarning: unclosed transport <_SelectorSocketTransport fd=26>
  source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/asyncio/selector_events.py:658: ResourceWarning: unclosed transport <_SelectorSocketTransport fd=33>
  source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/asyncio/selector_events.py:658: ResourceWarning: unclosed transport <_SelectorSocketTransport fd=34>
  source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/asyncio/selector_events.py:658: ResourceWarning: unclosed transport <_SelectorSocketTransport fd=28>
  source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/asyncio/selector_events.py:658: ResourceWarning: unclosed transport <_SelectorSocketTransport fd=29>
  source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/asyncio/selector_events.py:658: ResourceWarning: unclosed transport <_SelectorSocketTransport fd=30>
  source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/asyncio/selector_events.py:658: ResourceWarning: unclosed transport <_SelectorSocketTransport fd=31>
  source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/asyncio/selector_events.py:658: ResourceWarning: unclosed transport <_SelectorSocketTransport fd=32>
  source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.7/subprocess.py:883: ResourceWarning: subprocess 62307 is still running
  ResourceWarning, source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
./usr/lib/python3.7/subprocess.py:883: ResourceWarning: subprocess 62420 is still running
  ResourceWarning, source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
./usr/lib/python3.7/subprocess.py:883: ResourceWarning: subprocess 62530 is still running
  ResourceWarning, source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
./usr/lib/python3.7/subprocess.py:883: ResourceWarning: subprocess 62641 is still running
  ResourceWarning, source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
./usr/lib/python3.7/subprocess.py:883: ResourceWarning: subprocess 62644 is still running
  ResourceWarning, source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
./usr/lib/python3.7/subprocess.py:883: ResourceWarning: subprocess 62648 is still running
  ResourceWarning, source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
./usr/lib/python3.7/subprocess.py:883: ResourceWarning: subprocess 62651 is still running
  ResourceWarning, source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
./usr/lib/python3.7/subprocess.py:883: ResourceWarning: subprocess 62655 is still running
  ResourceWarning, source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback

----------------------------------------------------------------------
Ran 14 tests in 19.635s

OK
Setting up test:
	thread: 3
	queue size: 7
	policy: dh
Requesting url: http://localhost:8080/output.cgi
Requests succeeded: 7
Requests failed: 14
Setting up test:
	thread: 3
	queue size: 7
	policy: dh
Requesting url: http://localhost:8080/output.cgi
Requests succeeded: 7
Requests failed: 1
Setting up test:
	thread: 3
	queue size: 16
	policy: random
Requesting url: http://localhost:8080/output.cgi
Requests succeeded: 16
Requests failed: 8
Setting up test:
	thread: 3
	queue size: 16
	policy: random
Requesting url: http://localhost:8080/output.cgi
Requests succeeded: 16
Requests failed: 0
Setting up test:
	thread: 3
	queue size: 16
	policy: random
Requesting url: http://localhost:8080/output.cgi
Requests succeeded: 13
Requests failed: 4
Setting up test:
	thread: 3
	queue size: 7
	policy: dt
Requesting url: http://localhost:8080/output.cgi
Requests succeeded: 7
Requests failed: 14
Setting up test:
	thread: 3
	queue size: 7
	policy: dt
Requesting url: http://localhost:8080/output.cgi
Requests succeeded: 7
Requests failed: 1
Setting up test:
	thread: 3
	queue size: 80
	policy: dt
Requesting url: http://localhost:8080/output.cgi
Requests succeeded: 80
Requests failed: 0
Setting up test:
	thread: 9
	queue size: 80
	policy: dt
Requesting url: http://localhost:8080/output.cgi
Requests succeeded: 80
Requests failed: 0
Setting up test:
	thread: 3
	queue size: 80
	policy: dt
Requesting url: http://localhost:8080/output.cgi
Requests succeeded: 80
Requests failed: 0
Setting up test:
	thread: 1
	queue size: 1
	policy: dt
Setting up test:
	thread: 1
	queue size: 1
	policy: dt
Setting up test:
	thread: 1
	queue size: 1
	policy: dt
Setting up test:
	thread: 1
	queue size: 1
	policy: dt
Setting up test:
	thread: 1
	queue size: 1
	policy: dt
